pelicun/model/asset_model.py

```diff
@@ -223,6 +223,19 @@ def load_cmp_sample(self, filepath: str) -> None:
 
         self.cmp_units = units.groupby(level=0).first()
 
+        # Add marginal parameters with Blocks information (later calls
+        # rely on that attribute being defined)
+        # Obviously we can't trace back the distributions and their
+        # parameters, those columns are left undefined.
+        cmp_marginal_params = pd.DataFrame(
+            self.cmp_sample.columns.to_list(), columns=self.cmp_sample.columns.names
+        ).astype(str)
+        cmp_marginal_params['Blocks'] = 1
+        cmp_marginal_params = cmp_marginal_params.set_index(
+            ['cmp', 'loc', 'dir', 'uid']
```

Comment:
commit: https://github.com/NHERI-SimCenter/pelicun/pull/51/commits/b1e79bb8882ed23208bbd442f6542b03b05b6f87

We should prepare for the possibility of users providing a component sample without a 'uid' level in the index. Should be fairly easy and it is a better approach - in my opinion - than forcing everyone to add an extra level for no good reason.

Reply:
Great point. 2c9c38b0

Reply:
Good job not only adding the non-uid option but also taking care of some additional error-handling.